ecosystem/sf-indexer/aptos-sf-indexer/migrations/2022-08-08-043603_core_tables/up.sql

```diff
@@ -0,0 +1,285 @@
+  multi_sig_index BIGINT NOT NULL,
+  transaction_block_height BIGINT NOT NULL,
+  signer VARCHAR(255) NOT NULL,
+  is_sender_primary BOOLEAN NOT NULL,
+  type VARCHAR(255) NOT NULL,
+  public_key VARCHAR(255) NOT NULL,
+  threshold BIGINT NOT NULL,
+  public_key_indices jsonb NOT NULL,
+  inserted_at TIMESTAMP NOT NULL DEFAULT NOW(),
+  -- Constraints
+  PRIMARY KEY (
+    transaction_version,
+    multi_agent_index,
+    multi_sig_index
+  ),
+  CONSTRAINT fk_transaction_versions FOREIGN KEY (transaction_version) REFERENCES transactions (version)
+);
+/** Ex:
+ {
+ "key": "0x0400000000000000000000000000000000000000000000000000000000000000000000000a550c18",
+ "sequence_number": "0",
+ "type": "0x1::reconfiguration::NewEpochEvent",
+ "data": {
+ "epoch": "1"
+ }
+ }
+ */
+CREATE TABLE events (
+  key VARCHAR(100) NOT NULL,
+  sequence_number BIGINT NOT NULL,
+  creation_number BIGINT NOT NULL,
+  account_address VARCHAR(100) NOT NULL,
+  transaction_version BIGINT NOT NULL,
+  transaction_block_height BIGINT NOT NULL,
+  type TEXT NOT NULL,
+  data jsonb NOT NULL,
+  inserted_at TIMESTAMP NOT NULL DEFAULT NOW(),
+  -- Constraints
+  PRIMARY KEY (key, sequence_number),
+  CONSTRAINT fk_transaction_versions FOREIGN KEY (transaction_version) REFERENCES transactions (version)
+);
+CREATE INDEX event_key_seq_type_index ON events (key, sequence_number, type);
+CREATE TABLE write_set_changes (
+  transaction_version BIGINT NOT NULL,
+  hash VARCHAR(255) NOT NULL,
+  transaction_block_height BIGINT NOT NULL,
+  type TEXT NOT NULL,
+  address VARCHAR(255) NOT NULL,
+  index BIGINT NOT NULL,
+  inserted_at TIMESTAMP NOT NULL DEFAULT NOW(),
+  -- Constraints
+  PRIMARY KEY (transaction_version, hash),
+  CONSTRAINT fk_transaction_versions FOREIGN KEY (transaction_version) REFERENCES transactions (version)
+);
+CREATE INDEX write_set_changes_tx_hash_addr_type_index ON write_set_changes (transaction_version, address, type);
+CREATE TABLE move_modules (
+  transaction_version BIGINT NOT NULL,
+  write_set_change_index BIGINT NOT NULL,
+  transaction_block_height BIGINT NOT NULL,
+  name VARCHAR(255) NOT NULL,
+  address VARCHAR(255) NOT NULL,
+  bytecode text,
+  friends jsonb,
+  exposed_functions jsonb,
+  structs jsonb,
+  is_deleted BOOLEAN NOT NULL,
+  inserted_at TIMESTAMP NOT NULL DEFAULT NOW(),
+  -- Constraints
+  PRIMARY KEY (transaction_version, write_set_change_index),
+  CONSTRAINT fk_transaction_versions FOREIGN KEY (transaction_version) REFERENCES transactions (version)
+);
+CREATE TABLE move_resources (
+  transaction_version BIGINT NOT NULL,
+  write_set_change_index BIGINT NOT NULL,
+  transaction_block_height BIGINT NOT NULL,
+  name VARCHAR(255) NOT NULL,
+  address VARCHAR(255) NOT NULL,
+  module VARCHAR(255) NOT NULL,
+  generic_type_params jsonb,
+  data jsonb,
+  is_deleted BOOLEAN NOT NULL,
+  inserted_at TIMESTAMP NOT NULL DEFAULT NOW(),
+  -- Constraints
+  PRIMARY KEY (transaction_version, write_set_change_index),
```

Comment:
similar to above, something like `(address, module, name, transaction_version)` should also be globally unique and queryable

but lets talk about how to track this; would be helpful to see an example as json to better understand what we're getting into